src/main/java/com/replicon/bookTimeOff/MyTimeOffPage.java

```diff
@@ -0,0 +1,142 @@
+    @FindBy(css = "i[class='material-icons iconAddBooking']")
+    protected WebElement plusIcon;
+    @FindBy(css = "button[id='bookTimeOffBtn']")
+    protected WebElement timeOffButton;
+    @FindBy(css = "li[id='calenderTab'] > a")
+    protected WebElement calendar;
+    @FindBy(css = "table[class='fc-border-separate'] > tbody > tr:nth-child(2) > td:nth-child(2) > div > div:nth-child(1)")
+    protected WebElement selectDateFromCalendar;
+
+    public MyTimeOffPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void bookTimeOffFromMyTimeOff() {
+        returnToMenu.click();
+        bookTimeOffSection.click();
+        bookTimeOffButton.click();
+        waitForElementToBeVisible(typeDropdown);
+        typeDropdown.click();
+        waitForElementToBeVisible(selectLeaveType);
+        selectLeaveType.click();
+        startDateField.click();
+        waitForElementToBeClickable(selectStartDate);
+        selectStartDate.click();
+        endDateField.click();
+        waitForElementToBeClickable(selectEndDate);
+        selectEndDate.click();
+        submitForApproval.click();
+        pause(3);
+    }
+
+    public int beforeCount() {
+        bookTimeOffSection.click();
+        allTimeOffBookings.click();
+        return Integer.parseInt(beforeCountOfAllRequests.getText());
+    }
+
+    public int afterCount() {
+        pause(3);
+        allTimeOffBookings.click();
+        return Integer.parseInt(beforeCountOfAllRequests.getText());
+    }
+
+    public int allTimeOffBookingsAfterCount() {
+        pause(3);
+        returnToMenu.click();
+        pause(2);
+        bookTimeOffSection.click();
+        allTimeOffBookings.click();
+        return Integer.parseInt(beforeCountOfAllRequests.getText());
+    }
+
+    public void bookTimeOffFromMyTimeOffByPlusIcon() {
+        returnToMenu.click();
+        bookTimeOffSection.click();
+        plusIcon.click();
+        waitForElementToBeVisible(typeDropdown);
+        typeDropdown.click();
+        waitForElementToBeVisible(leaveType);
+        leaveType.click();
+        startDateField.click();
+        waitForElementToBeClickable(selectStartDate);
+        selectStartDate.click();
+        endDateField.click();
+        waitForElementToBeClickable(selectEndDate);
+        selectEndDate.click();
+        submitForApproval.click();
+    }
+
+    public void bookTimeOffByAllTimeOffBookings() {
+        returnToMenu.click();
+        bookTimeOffSection.click();
+        allTimeOffBookings.click();
+        timeOffButton.click();
+        waitForElementToBeVisible(typeDropdown);
+        typeDropdown.click();
+        waitForElementToBeVisible(leaveType);
+        leaveType.click();
+        startDateField.click();
+        waitForElementToBeClickable(selectStartDate);
+        selectStartDate.click();
+        endDateField.click();
+        waitForElementToBeClickable(selectEndDate);
+        selectEndDate.click();
+        submitForApproval.click();
```

Comment:
Most of the methods have similar functionality and structure. To reduce code duplication and improve maintainability, we can combine the common logic into a single method.